Initialtask/src/Entity/Product.php

```diff
@@ -0,0 +1,194 @@
+ * @ORM\Entity(repositoryClass=ProductRepository::class)
+ */
+class Product
+{
+    public const MIN_COST = 5;
+
+    public const MAX_COST = 1000;
+
+    public const MIN_STOCK = 10;
+
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(name="intProductDataId", type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(name="strProductName", type="string", length=50)
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(name="strProductCode", type="string", length=10, unique=true)
+     */
+    private $code;
+
+    /**
+     * @ORM\Column(name="strProductDesc", type="string", length=255)
+     */
+    private $description;
+
+    /**
+     * @ORM\Column(name="intStock", type="integer", nullable=true)
+     */
+    private $stock;
+
+    /**
+     * @ORM\Column(name="floatCost", type="float", nullable=true)
+     */
+    private $cost;
+
+    /**
+     * @ORM\Column(name="dtmDiscontinued", type="datetime", nullable=true)
+     */
+    private $discontinued;
+
+    /**
+     * @ORM\Column(name="dtmAdded", type="datetime", nullable=true)
+     */
+    private $added;
+
+    /**
+     * @ORM\Column(name="stmTimestamp", type="datetime", nullable=true)
+     */
+    private $updated;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    private function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    public function getCode(): ?string
+    {
+        return $this->code;
+    }
+
+    private function setCode(string $code): self
+    {
+        $this->code = $code;
+
+        return $this;
+    }
+
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    private function setDescription(string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    public function getStock(): ?int
+    {
+        return $this->stock;
+    }
+
+    private function setStock(?int $stock): self
+    {
+        $this->stock = $stock;
+
+        return $this;
+    }
+
+    public function getCost(): ?float
+    {
+        return $this->cost;
+    }
+
+    private function setCost(?float $cost): self
+    {
+        $this->cost = $cost;
+
+        return $this;
+    }
+
+    public function getDiscontinued(): ?\DateTimeInterface
+    {
+        return $this->discontinued;
+    }
+
+    private function setDiscontinued(string $discontinued): self
+    {
+        if ('yes' === $discontinued) {
+            $this->discontinued = new \DateTimeImmutable('now');     //if discontinued setting current date
+        } else {
+            $this->discontinued = null;
+        }
+
+        return $this;
+    }
+
+    public function getAdded(): ?\DateTimeInterface
+    {
+        return $this->added;
+    }
+
+    private function setAdded(?\DateTimeInterface $added): self
```

Comment:
Please have a look at https://symfony.com/doc/4.1/doctrine/lifecycle_callbacks.html
we don't need to set it manually